src/main/scala/org/scalacheck/Gen.scala

```diff
@@ -258,10 +259,17 @@ object Gen extends GenArities{
 
     private def chDbl(l: Double, h: Double)(p: P, seed: Seed): R[Double] = {
       val d = h-l
-      if (d < 0 || d > Double.MaxValue) r(None, seed)
+      val (n, s) = seed.double
+      if (d < 0) r(None, seed)
+      else if (d > Double.MaxValue) {
+        if(Random.nextBoolean()){
```

Comment:
@jahangirmohammed Calling `Random.nextBoolean()` here is not deterministic. Given a `seed` value we want to always return the same result.